Fix opt parsing, clean UUID in path, allow additional tags

LucasRoesler opened the pull request:
**What**
- Fix option parsing for the tracing middleware
- Set the default operation name function to scrub UUID values from the
url path
- Allow passing additional fixed set of tags to the spans

**Why**
- This creates operation names that are easier to read and aggregate
- The additional tags allow us to pass additional serverwide values as
tags to the routing spans, e.g. tenantid or application version

**Notes**
- I didn't update the grpc middleware because I wasn't sure exactly how it was working, it didn't seem to have any real customization going on.  It would be nice to update this with the extra tags option that was added for the http server

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

LucasRoesler committed:
Fix opt parsing, clean UUID in path, allow additional tags

**What**
- Fix option parsing for the tracing middleware
- Set the default operation name function to scrub UUID values from the
url path
- Allow passing additional fixed set of tags to the spans

**Why**
- This creates operation names that are easier to read and aggregate
- The additional tags allow us to pass additional serverwide values as
tags to the routing spans, e.g. tenantid or application version

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

trusch reviewed (commented):
I don't know if I like the uuid feature of this PR. While undoubtedly helpfull for our usecase I wish I could just opt-in to that feature, or at least opt-out when I need the uuid.

LucasRoesler commented:
I would argue that it is highly unlikely that the Operation name should include a uuid. I agree that controlling this as an option is definitely the best though.

On the other hand, the package does support a generic  interface for constructing and passing additional middlewares as server options.

LucasRoesler committed:
Allow user configuration of opNameFunc

**What**
- Allow the user to pass a function to customize how the operation named
from the request

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

LucasRoesler commented:
@trusch i updated the PR to allow the opNameFunc to be configured by the user via the `WithTracing` method

LucasRoesler committed:
Export the MethodAndPathCleanID

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>
